### PR TITLE
#636 Automatically invalidate static resources on new deployment

### PR DESCRIPTION
**Pull request by lokm01**
- uses the fixed version resolver in Spring
- index first gets the version from the API
     - then programmatically loads all the dependencies with the
        version prefix
     - only on load of the last dependency do we initialize the
        component container (which is configured to load all
        local resources with the version prefix)

This changes requires the 3 new properties in application
properties.

@GeorgiChochov any idea if the webjar locator can be configured for the versioned resources somehow?

### CODE REVIEW

**Comment by GeorgiChochov** [comment]
No idea if the webjar locator can be configured for versioned files. Do we really need to version them? They won't be changing with significant frequency. If really necessary, we might skip the locator and pass in a full path with I think contains the package's version.

**Comment by lokm01** [comment]
> No idea if the webjar locator can be configured for versioned files. Do we really need to version them? They won't be changing with significant frequency. If really necessary, we might skip the locator and pass in a full path with I think contains the package's version.

Cool.

File: menas/ui/components/app.controller.js [hunk 1]
```diff
@@ -33,6 +33,13 @@ sap.ui.define([
 
       this._app = this.byId("menasApp");
 
+      // register TNT icons
+      sap.ui.core.IconPool.registerFont({
```
Comment (GeorgiChochov) [inline menas/ui/components/app.controller.js]: What's this for?

Comment (lokm01) [reply menas/ui/components/app.controller.js]: This registers the namespace for technical (TNT) icons.

https://openui5.hana.ondemand.com/test-resources/sap/m/demokit/iconExplorer/webapp/index.html#/overview/SAP-icons-TNT

File: menas/ui/index.html [hunk 2]
```diff
@@ -20,65 +20,88 @@
...
     <script>
-        sap.ui.localResources("components");
-        
-        // register TNT icons
-        sap.ui.core.IconPool.registerFont({
-          collectionName: "SAP-icons-TNT",
-          fontFamily: "SAP-icons-TNT", 
-          fontURI: sap.ui.require.toUrl("sap/tnt/themes/base/fonts"),
-          lazy: true});
+      function _menasLoadElem(sElemType, sSrc, aAttrs, fnOnLoad) {
```
[the remark this answers is not in the document]
Comment (lokm01) [reply menas/ui/index.html]: Hmm the problem is that this is the code which loads all the common stuff :)

Btw defined as global function so should be accessible everywhere. Alternatively we could attach it to window for now (not sure what's cleaner), to be honest I'm not super happy about this change in general.

File: menas/ui/index.html [hunk 3]
```diff
@@ -17,72 +17,95 @@
 
 <html>
 <head>
-    <meta http-equiv="X-UA-Compatible" content="IE=edge">
-    <meta http-equiv='Content-Type' content='text/html;charset=UTF-8'/>
+<meta http-equiv="X-UA-Compatible" content="IE=edge">
```
[the remark this answers is not in the document]
Comment (lokm01) [reply menas/ui/index.html]: Comes from templates I guess..

File: menas/ui/components/app.controller.js [hunk 4]
```diff
@@ -118,10 +125,6 @@ sap.ui.define([
 
     onLogout: function () {
       this._router.navTo("root");
-    },
-
-    onAfterRendering: function () {
```
[the remark this answers is not in the document]
Comment (lokm01) [reply menas/ui/components/app.controller.js]: I think the idea was to make it busy while the components were loading. I don't think this ever worked properly, we should rather make the specific components busy as they make service requests (example is implemented in scheduling)

**Pull request merged**